File: builtin/string.py
from .number import Number
from .boolean import Boolean

class String:
    def __init__(self, string):
        self.str = self.value = string
        self.length = Number(len(self.str))

    def __str__(self):
        return self.value

    def toNumber(self):
        return float(self.str) if '.' in self.str else int(self.str)

    def standsFor(self, str, visit):
        return Boolean((visit.visit(str[0]).value).startswith(self.str))

File: builtin/string.js
const Numberr = require('./number')
const Booleann = require('./boolean')

module.exports = class Stringg {
  constructor (str) {
    this.str = this.value = str
  }

  toString () {
    return this.value
  }

  toNumber () {
    return this.str.includes('.') ? parseFloat(this.str) : parseInt(this.str)
  }

  get length () {
    return new Numberr(this.str.length)
  }

  standsFor (str, visit) {
    return new Booleann((visit.visit(str[0]).value).startsWith(this.str))
  }
}
